makefile.shade

```diff
@@ -203,15 +204,176 @@ var buildTarget = "compile"
+                        var newReplaceExp = string.IsNullOrEmpty(newRoslynPrerelease) ? "-*" : 
+                                            stableBranches.Contains(branch) ? string.Format("-{0}", newRoslynPrerelease) : string.Format("-{0}-*", newRoslynPrerelease);
+
+                        builder.AppendLine(jsonEntry.Replace(oldSearchExp, newReplaceExp));
+                    }
+                    else if(jsonEntry.Contains("IdentityModel"))
+                    {
+                        // IdentityModel packages don't have a stable build and are updated very frequently on the NuGet feed
+                        builder.AppendLine(jsonEntry);
+                    }
+                    else
+                    {
+                        var oldSearchExp = string.IsNullOrEmpty(oldPreRelease) ? "-*" : string.Format("-{0}-*", oldPreRelease);
+                        var newReplaceExp = string.IsNullOrEmpty(newPreRelease) ? "-*" : 
+                                            stableBranches.Contains(branch) ? string.Format("-{0}", newPreRelease) : string.Format("-{0}-*", newPreRelease);
+
+                        var newEntry = ProcessExcludesForPackages(jsonEntry, stableBranches.Contains(branch));
+
+                        builder.AppendLine(newEntry.Replace(oldSearchExp, newReplaceExp));
+                    }
+                }
+
+                File.WriteAllText(file,builder.ToString());
+            }
+
+            // Build projects to produce project.lock.json
+            Exec("build.cmd", "compile", repo);
+
+            // Replace lock=true in project.lock.json
+            foreach(var file in Directory.GetFiles(repo, "project.lock.json", SearchOption.AllDirectories))
+            {
+                var contents = File.ReadAllText(file);
+                var newContents = contents.Replace("\"locked\": false", "\"locked\": true");
+                File.WriteAllText(file, newContents);
+            }
+
+            // Add the project.json and project.lock.json files
+            foreach(var file in Directory.GetFiles(repo, "project*", SearchOption.AllDirectories))
+            {
+                var gitFilePath = file.Substring(file.IndexOf(repo) + repo.Length + 1);
+
+                // Check if this needs to be included
+                if(ShouldIncludeProjectJson(gitFilePath))
+                {
+                    GitCommand(repo, string.Format("add -f {0}", gitFilePath));
+                }
+            } 
+
+            GitCommand(repo, "commit -am \"Updating json files to pin versions and build.cmd to pin KoreBuild and DNX\""); 
+
+            GitCommand(repo, string.Format("push origin {0}:{0} -f", branch));
+        }
+    }
+
+#update-prerelease-tags
+    @{
+        var newPreRelease = Environment.GetEnvironmentVariable("PRERELEASETAG");
+        if(string.IsNullOrEmpty(newPreRelease))
+        {
+            Log.Warn("No prerelease tag defined");
```

Comment:
Can we add little instructions here on how to run these? E.g. "set env var BLAH to [something], then run foo_bar with bar param"?